tests/modules/text_generation/test_text_generation_local.py

```diff
@@ -211,7 +212,26 @@ def test_zero_epoch_case(disable_wip):
     assert isinstance(model.model, HFAutoSeq2SeqLM)
 
 
-def test_run_tokenizer_not_implemented():
-    with pytest.raises(NotImplementedError):
-        model = TextGeneration.bootstrap(SEQ2SEQ_LM_MODEL)
-        model.run_tokenizer("This text doesn't matter")
+# ############################## Run Tokenizer ################################
+
+
+def test_run_tokenizer_edge_cases(disable_wip, set_cpu_device):
+    """Test tokenizer on edge cases like empty strings and long input."""
+    model = TextGeneration.bootstrap(CAUSAL_LM_MODEL)
+
+    # Edge case: Empty string
+    empty_result = model.run_tokenizer("")
+    assert isinstance(empty_result, TokenizationResults)
+    assert empty_result.token_count == 0
+
+    # Normal case: short sentence
+    short_text = "This is a test sentence."
+    short_result = model.run_tokenizer(short_text)
+    assert isinstance(short_result, TokenizationResults)
+    assert short_result.token_count > 0
```

Comment:
Checking for an expected number instead of or in addition to a non-zero condition seems like a good idea to increase the robustness of the test. 

What are your thoughts on re-writing such a test like this:
```
assert short_result.token_count == len(model.model.tokenizer.encode(short_text))
```

Reply:
Great -- changes to tests have been implemented accordingly and they seem to pass